### PR TITLE
Remove Typechain hacks, avoid using sed

### PR DESCRIPTION
**Pull request by Siegrift**
Fixes https://api3workspace.slack.com/archives/C01N6001YCE/p1621963547042900

The hacks I needed to use for TypeChain were probably caused by dependency clashed, which now seems to be resolved, but I had to add the generated files to eslintignore (not sure why).

Also avoided the using `sed`, because it is not cross platform compatible (although for that specific use case it probably was).

### CODE REVIEW

File: packages/protocol/dev-scripts/delete-flattened-licenses.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const replace = require('replace-in-file');
+
+const results = replace.sync({
+  files: '*.flat.sol',
+  from: /.*SPDX-License-Identifier.*\n/g,
```
Comment (andreogle) [inline packages/protocol/dev-scripts/delete-flattened-licenses.js:5]: Why are we removing this "license-identifier"? I'm just curious

**Review by andreogle — APPROVED** [review]

File: packages/protocol/dev-scripts/delete-flattened-licenses.js (hunk 1 above)
Comment (Siegrift) [reply packages/protocol/dev-scripts/delete-flattened-licenses.js:5]: I don't know :)

Comment (andreogle) [reply packages/protocol/dev-scripts/delete-flattened-licenses.js:5]: But things break without removing it?

Comment (Siegrift) [reply packages/protocol/dev-scripts/delete-flattened-licenses.js:5]: No, I guess this is unrelated to core development. Maybe if you want to copy and modify the contract? Not really sure, I think @bbenligiray would know the answer.

**Pull request merged**